File: number_guessing_game.py
# Tag eins

# This is the number guessing game
import random

while True:
    status = True
    random_number = random.randint(1, 100)
    print("\nErrate die Zahl, welche zwischen 1 und 100 liegt: ")

    while status:
        guess = int(input("Gib eine Zahl ein: "))

        if guess == random_number:
            print("Glückwunsch! Ihr hast die Zahl erraten.\n")
            status = False
        elif guess < random_number:
            print("Zu niedrig!")
        elif guess > random_number:
            print("Zu hoch!")
        else:
            print("Ungültige Eingabe")

        

File: number_guessing_game.c


#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <unistd.h>
#include <stdbool.h>

int main() {
    unsigned int seed = time(NULL) ^ getpid();
    srand(seed);

    int randomNumber = rand() % 100 + 1;
    int guess;
    bool status = true;

    while (status) {
        printf("Errate die Zahl zwischen 1 und 100: ");
        
        // Check if the input is a valid integer
        if (scanf("%d", &guess) != 1) {
            printf("Fehler, bitte geben Sie eine Zahl zwischen 1 und 100 ein!\n");
            // Clear the input buffer
            while (getchar() != '\n');
            continue;
        }

        if (guess == randomNumber) {
            printf("Super, ihr habt die Zahl erraten!\n");
            status = false;
        } else if (guess < randomNumber) {
            printf("Zu niedrig, versucht es nochmal!\n");
        } else if (guess > randomNumber) {
            printf("Zu hoch, versucht es nochmal!\n");
        } else {
            printf("Fehler, bitte geben Sie eine Zahl zwischen 1 und 100 ein!\n");
        }
    }

    return 0;
}

